src/Windows/R/Editor/Impl/Selection/SelectWordCommand.cs

```diff
@@ -20,13 +24,19 @@ public sealed class SelectWordCommand : ViewCommand {
 
         public override CommandResult Invoke(Guid group, int id, object inputArg, ref object outputArg) {
             if (!TextView.Caret.InVirtualSpace) {
-                int caretPosition = TextView.Caret.Position.BufferPosition.Position;
-                var snapshot = TextView.TextBuffer.CurrentSnapshot;
-                Span? spanToSelect = RTextStructure.GetWordSpan(snapshot, caretPosition);
-                if (spanToSelect.HasValue && spanToSelect.Value.Length > 0) {
-                    TextView.Selection.Select(new SnapshotSpan(snapshot, spanToSelect.Value), isReversed: false);
-                    TextView.Caret.MoveTo(new SnapshotPoint(snapshot, spanToSelect.Value.End));
-                    return CommandResult.Executed;
+                var point = TextView.MapDownToBuffer(TextView.Caret.Position.BufferPosition, _textBuffer);
+                if (point.HasValue) {
+                    var snapshot = point.Value.Snapshot;
+                    var spanToSelect = RTextStructure.GetWordSpan(snapshot, point.Value);
+                    if (spanToSelect.HasValue && spanToSelect.Value.Length > 0) {
+                        var start = TextView.MapUpToView(snapshot, spanToSelect.Value.Start);
```

Comment:
Don't you want to make it an extension method?
`MapUpToView(this ITextView textView, ITextSnapshot snapshot, Span span)`